lib/server-builder.js

```diff
@@ -0,0 +1,124 @@
+import { SubProcess } from 'teen_process';
+import {fs, logger, system} from 'appium-support';
+import log from './logger';
+import path from 'path';
+import { EOL } from 'os';
+
+const IGNORED_ERRORS = [];
+const GRADLE_VERSION_KEY = 'gradle_version';
+const GRADLE_URL_PREFIX = 'distributionUrl=';
+const GRADLE_URL_TEMPLATE = 'https\\://services.gradle.org/distributions/gradle-VERSION-all.zip';
+const GRADLE_MAX_ERROR_LOG_LINES = 15;
+
+const VERSION_KEYS = [
+  GRADLE_VERSION_KEY,
+  'android_gradle_plugin_version',
+  'compile_sdk_version',
+  'build_tools_version',
+  'min_sdk_version',
+  'target_sdk_version',
+  'kotlin_version'
+];
+
+const gradleLog = logger.getLogger('Gradle');
+
+class ServerBuilder {
+  constructor (args = {}) {
+    this.serverPath = args.serverPath;
+    this.showGradleLog = args.showGradleLog;
+
+    const versionsArg = args.versions || {};
+    const serverVersions = {};
+    for (const key of VERSION_KEYS) {
+      if (versionsArg[key]) {
+        serverVersions[key] = versionsArg[key];
+      }
+    }
+    this.server_versions = serverVersions;
+  }
+
+  async build () {
+    if (this.server_versions[GRADLE_VERSION_KEY]) {
+      await this.setGradleWrapperVersion(this.server_versions[GRADLE_VERSION_KEY]);
+    }
+    await this.runBuildProcess();
+  }
+
+  getCommand () {
+    let cmd = system.isWindows() ? 'gradlew.bat' : './gradlew';
```

Comment:
👍

Reply:
const

Reply:
Also I assume that `./` prefix is not needed for gradlew

Reply:
As I understand it, `./` prefix makes sure that gradlew will be started from the passed working directory, not from somewhere else in PATH. I think this would be more clean than going without `./`.

Reply:
not really. for a normal command prompt adding `./` prefix requests to execute the file
For subprocess having `cwd` option set is enough though

Reply:
Unfortunately just `gradlew` doesn't work on my mac, so I'll keep it `./gradlew`. For some reason it tries to run `gradlew` from PATH if there is no `./`.

Reply:
hm, ok. Thanks for checking it